core/src/main/java/org/apache/james/mime4j/stream/RawFieldParser.java

```diff
@@ -191,6 +195,12 @@ public String parseToken(final ByteSequence buf, final ParserCursor cursor, fina
      *  is not delimited by any character.
      */
     public String parseValue(final ByteSequence buf, final ParserCursor cursor, final BitSet delimiters) {
+        if (!CharsetUtil.isASCII(buf)) {
+            String value = parseUtf8Filename(buf);
```

Comment:
So every time we parse a raw field we look ahead to see if it is a finename?

This do not look performant to me. I can see in the the future more and more `if` being added in there.

The mime4J logic tends more to be to delay parsing.

I would way prefer having helper methods in the ContentDisposition field (and other fields as well) doing Q decoding just if the user wants it (`contentDisposition.getDecodedFilename()`)

Reply:
I guess you meant UTF-8 decoding here?